diff --git a/packages/sherpa-onnx.rn/android/src/main/kotlin/com/k2fsa/sherpa/onnx/OnlineRecognizer.kt b/packages/sherpa-onnx.rn/android/src/main/kotlin/com/k2fsa/sherpa/onnx/OnlineRecognizer.kt
index 8ad93ac8..c80caec6 100644
--- a/packages/sherpa-onnx.rn/android/src/main/kotlin/com/k2fsa/sherpa/onnx/OnlineRecognizer.kt
+++ b/packages/sherpa-onnx.rn/android/src/main/kotlin/com/k2fsa/sherpa/onnx/OnlineRecognizer.kt
@@ -84,6 +84,7 @@ data class OnlineRecognizerResult(
     val text: String,
     val tokens: Array<String>,
     val timestamps: FloatArray,
+    // MODIFIED(v1.12.28): added ysProbs field to match JNI ABI; upstream was missing this field
     val ysProbs: FloatArray,
     // TODO(fangjun): Add more fields
 )
